src/test/scala/chiselTests/util/experimental/minimizer/MinimizerSpec.scala

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.util.experimental.minimizer
+
+import chisel3._
+import chisel3.util._
+import chisel3.util.experimental.decode._
+import chisel3.util.pla
+import chiselTests.SMTModelCheckingSpec
+
+class DecodeTestModule(minimizer: Minimizer, table: TruthTable) extends Module {
+  val i = IO(Input(UInt(table.table.head._1.getWidth.W)))
+  val (unminimizedI, unminimizedO) = pla(table.table.toSeq)
+  unminimizedI := i
+  val minimizedO: UInt = decoder(minimizer, i, table)
+
+  chisel3.experimental.verification.assert(
+    // for each instruction, if input matches, output should match, not no matched, fallback to default
+    (table.table.map { case (key, value) => (i === key) && (minimizedO === value) } ++
+      Seq(table.table.keys.map(i =/= _).reduce(_ && _) && minimizedO === table.default)).reduce(_ || _)
+  )
+}
+
+trait MinimizerSpec extends SMTModelCheckingSpec {
+  def minimizer: Minimizer
+
+  def minimizerTest(testcase: TruthTable, caseName: String) = {
+    test(
+      () => new DecodeTestModule(minimizer, table = testcase),
+      s"${minimizer.getClass.getSimpleName}.$caseName",
+      success
+    )
+  }
+
+  "case0" should "pass" in {
+    minimizerTest(TruthTable(
+      Map(
+        // BitPat("b000") -> BitPat("b0"),
+        BitPat("b001") -> BitPat("b?"),
+        BitPat("b010") -> BitPat("b?"),
+        // BitPat("b011") -> BitPat("b0"),
+        BitPat("b100") -> BitPat("b1"),
+        BitPat("b101") -> BitPat("b1"),
+        // BitPat("b110") -> BitPat("b0"),
+        BitPat("b111") -> BitPat("b1")
+      ),
+      BitPat("b0")
```

Comment:
What are the comments for? Is it indicating that when you leave a pattern off, it defaults to requiring the output to be 0?

Reply:
> What are the comments for? Is it indicating that when you leave a pattern off, it defaults to requiring the output to be 0?

it defaults to requiring the output to be 0?
Leave this to default to 0 and optimized by decode algorithm.

Reply:
I’ll add more comments here tonight.